DataCapturing/DataCapturingService.swift

```diff
@@ -92,6 +95,9 @@ public class DataCapturingService: NSObject {
     /// A timer called in regular intervals to save the captured data to the underlying database.
     private var backgroundSynchronizationTimer: DispatchSourceTimer!
 
+    /// The number of the current event. This is used to filter events based on `locationUpdateRate`.
+    private var geoLocationEventNumber = 0
```

Comment:
this is a counter for how many geo location update events have already been skipped. It is set to 0 when it reaches locationUpdateSkipRate